### PR TITLE
Replace keys with safe values when to_s fails

### PR DESCRIPTION
**Pull request by imjoehaines**
## Goal

This PR fixes a issue with https://github.com/bugsnag/bugsnag-ruby/pull/591 where we filter the value of a hash if its key raises, but we leave the key alone. That means that code later on can cause the same issue we were trying to fix — if they run `to_s` on the key then it will raise or infinitely recurse

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: lib/bugsnag/cleaner.rb [hunk 1]
```diff
@@ -29,10 +29,19 @@ def traverse_object(obj, seen, scope)
...
+            else
+              clean_hash[k] = traverse_object(v, seen, [scope, k].compact.join('.'))
+            end
+          # If we get an error here, we assume the key needs to be filtered
+          # to avoid leaking things we shouldn't. We also remove the key itself
+          # because it may cause issues later e.g. when being converted to JSON
+          rescue StandardError
+            clean_hash[RAISED] = FILTERED
+          rescue SystemStackError
+            clean_hash[RECURSION] = FILTERED
```
Comment (imjoehaines) [inline lib/bugsnag/cleaner.rb:44]: Reusing the same key here does mean we'd overwrite it if more than one key throws, but I think that's unlikely to happen realistically and I don't think you'd get any useful information out of `[RAISED 1]... [RAISED 2]` if we made them unique
